src/test/components/MarkerChart.test.js

```diff
@@ -380,21 +389,56 @@ describe('MarkerChart', function() {
         getContextMenu,
         findFillTextPosition,
         clickOnMenuItem,
+        clickOnSubMenuItem,
       };
     }
 
     it('displays when right clicking on a marker', () => {
       const {
         rightClick,
         clickOnMenuItem,
+        clickOnSubMenuItem,
         getContextMenu,
         findFillTextPosition,
+        getState,
       } = setupForContextMenus();
 
       rightClick(findFillTextPosition('UserTiming A'));
 
       expect(getContextMenu()).toHaveClass('react-contextmenu--visible');
 
+      clickOnSubMenuItem('From the start of this marker', 0);
+      expect(getPreviewSelection(getState())).toEqual({
+        hasSelection: true,
+        isModifying: false,
+        selectionEnd: 11,
+        selectionStart: 0,
+      });
+
+      clickOnSubMenuItem('From the end of this marker', 0);
+      expect(getPreviewSelection(getState())).toEqual({
+        hasSelection: true,
+        isModifying: false,
+        selectionEnd: 11,
+        selectionStart: 10,
+      });
+
+      clickOnSubMenuItem('From the start of this marker', 1);
+      expect(getPreviewSelection(getState())).toEqual({
+        hasSelection: true,
+        isModifying: false,
+        selectionEnd: 11,
+        selectionStart: 10,
+      });
+
+      clickOnSubMenuItem('From the end of this marker', 1);
+      expect(getPreviewSelection(getState())).toEqual({
+        hasSelection: true,
+        isModifying: false,
+        selectionEnd: 10.0001,
+        selectionStart: 10,
+      });
+
```

Comment:
This `it` testing block check if the context menu displays when right click on a marker. Testings should be isolated and their text should be descriptive so people can understand what's failing/passing when they are analyzing the tests. Can you please move this context menu clicking test logic to its own test block with a descriptive name?

Reply:
Yea sure.